src/headless/plugins/blocking/api.js

```diff
@@ -4,97 +4,103 @@ import { _converse, api, converse } from '@converse/headless/core.js';
 const { Strophe, $iq, sizzle, u } = converse.env;
 
 export default {
+
+    /**
+     * Checks if XEP-0191 is supported
+     */
+    async isBlockingAvailable () {
+        const has_feature = await api.disco.supports(Strophe.NS.BLOCKING, _converse.domain);
+        if (!has_feature) {
+            log.info("XEP-0191 not supported, no blocklist available");
+            return false;
+        }
+        log.debug("XEP-0191 available");
+        return true;
+    },
     /**
      * Retrieves the blocklist held by the logged in user at a JID by sending an IQ stanza.
-     * Saves the model variable _converse.blocked.set
      * @private
      * @method api.refreshBlocklist
      */
     async refreshBlocklist () {
-        const features = await api.disco.getFeatures(_converse.domain);
-        if (!features?.findWhere({ 'var': Strophe.NS.BLOCKING })) {
-            return false;
+        log.debug("refreshing blocklist");
+        const available = await this.isBlockingAvailable();
+        if (!available) {
+            log.debug("XEP-0191 NOT available, not refreshing...");
+            api.trigger('blockListFetched', []);
```

Comment:
Is triggering this event necessary here? Strictly speaking, I don't think it's correct, because the block list wasn't fetched (since blocking isn't supported).

Looks like you should also `return` in this `if` statement.